Server/ConcordiaCurriculumManager/Models/Curriculum/Dossiers/CourseCreationRequest.cs

```diff
@@ -0,0 +1,16 @@
+﻿using ConcordiaCurriculumManager.Models.Users;
+
+namespace ConcordiaCurriculumManager.Models.Curriculum.Dossiers;
+
+public class CourseCreationRequest : BaseModel
+{
+    public Guid InitiatorId { get; set; }
```

Comment:
The NewCourseId, and DossierId should be required. The Course type should be Course? and the Dossier type should be Dossier?. The InitiatorId field should be removed. Afterwards, you can generate a new migration to confirm it works.